src/node/rpc/frontend.h

```diff
@@ -455,18 +455,27 @@ namespace ccf
         {
           case HandlerRegistry::Read:
           {
+            if (ctx->session.is_forwarded)
+            {
+              return forward_or_redirect_json(ctx);
+            }
             break;
           }
 
           case HandlerRegistry::Write:
           {
+            ctx->session.is_forwarded = true;
             return forward_or_redirect_json(ctx);
-            break;
           }
 
           case HandlerRegistry::MayWrite:
           {
             if (!ctx->read_only_hint)
```

Comment:
Can this become a single if statement?

Reply:
It can, and I did consider it, but setting is_forwarded right after checking for it seemed... awkward? I'm happy to make the change though, I don't feel stronly about it.